/*
 * This file was automatically generated by EvoSuite
 */

package org.jbehave.core.failures;

import org.junit.Test;
import org.junit.runner.RunWith;
import static org.junit.Assert.*;
import org.jbehave.core.failures.RethrowingFailure;
import org.jbehave.core.failures.UUIDExceptionWrapper;


public class RethrowingFailureEvoSuiteTest {


  //Test case number: 0
  /*
   * 2 covered goals:
   * 1 org.jbehave.core.failures.RethrowingFailure.<init>()V: root-Branch
   * 2 org.jbehave.core.failures.RethrowingFailure.handleFailure(Ljava/lang/Throwable;)V: I4 Branch 1 IFEQ L6 - true
   */
  @Test
  public void test0()  throws Throwable  {
      RethrowingFailure rethrowingFailure0 = new RethrowingFailure();
      Throwable throwable0 = new Throwable("");
      try {
        rethrowingFailure0.handleFailure(throwable0);
      }catch(Throwable e) {
        /*
         * 
         */
        /*
      %%%<ex><className val="Throwable"/><stackDepth val="13"/><causeDepth val="0"/><source val="sun.reflect.GeneratedConstructorAccessor9.newInstance(Unknown Source)"/></ex>%%%
         */
      }
  }

  //Test case number: 1
  /*
   * 2 covered goals:
   * 1 org.jbehave.core.failures.RethrowingFailure.<init>()V: root-Branch
   * 2 org.jbehave.core.failures.RethrowingFailure.handleFailure(Ljava/lang/Throwable;)V: I4 Branch 1 IFEQ L6 - false
   */
  @Test
  public void test1()  throws Throwable  {
      RethrowingFailure rethrowingFailure0 = new RethrowingFailure();
      UUIDExceptionWrapper uUIDExceptionWrapper0 = new UUIDExceptionWrapper();
      try {
        rethrowingFailure0.handleFailure((Throwable) uUIDExceptionWrapper0);
      }catch(NullPointerException e) {
        /*
         * no message in exception (getMessage() returned null)
         */
        /*
      %%%<ex><className val="NullPointerException"/><stackDepth val="14"/><causeDepth val="0"/><source val="org.jbehave.core.failures.RethrowingFailure.handleFailure(RethrowingFailure.java:7)"/></ex>%%%
         */
      }
  }
}
